### ISSUE TITLE
Ship of Harkinian and Ship of Harkinian 2

### ISSUE THREAD

**Issue by sergiogbrox** [cizia64/CrossMix-OS#291]
I would like to thank you for the great work you have been doing. Thank you!!

My problem: Today I tried to install the Legend of Zelda Ocarina of Time and Majoras Mask port (Ship of Harkinian and Ship of Harkinian 2) via port master. I saw a tutorial on YouTube from a guy saying that I had to install Knulli for it to work, but I thought that even so I should try. After the attempt I saw that it really didn't work. I tried to read the logs but I'm very ignorant so I would like to know if this would be possible to solve in a system update or unfortunately it only runs on Knulli itself.

Log Ship of Harkinian:
/mnt/SDCARD/Emus/PORTS/../../Roms/PORTS/Ship of Harkinian.sh: line 40: can't open imgui.ini: no such file
mv: can't rename 'imgui_temp.ini': No such file or directory
Checking if CFW_NAME 'TrimUI' is in the list...
Comparing 'ArkOS' with 'TrimUI'...
Comparing 'ArkOS wuMMLe' with 'TrimUI'...
Comparing 'ArkOS AeUX' with 'TrimUI'...
Comparing 'TrimUI' with 'TrimUI'...
Match found: 'TrimUI'
Missing ROM files! Can't generate otr!
Loading, please wait... (might take a while!)
Running in UINPUT output mode.
./soh.elf: /usr/lib/libz.so.1: no version information available (required by /mnt/SDCARD/System/lib/libpng16.so.16)
./soh.elf: /usr/lib/libz.so.1: no version information available (required by /mnt/SDCARD/System/lib/libpng16.so.16)
./soh.elf: symbol lookup error: /mnt/SDCARD/System/lib/libpng16.so.16: undefined symbol: inflateValidate, version ZLIB_1.2.9
Running in Fake Keyboard mode
Using ConfigFile soh.gptk
Joystick 0 has game controller name 'Xbox 360 Controller'

Log Ship of Harkinian 2:

/mnt/SDCARD/Emus/PORTS/../../Roms/PORTS/Ship of Harkinian 2.sh: line 40: can't open imgui.ini: no such file
mv: can't rename 'imgui_temp.ini': No such file or directory
Loading, please wait... (might take a while!)
Running in UINPUT output mode.
./2s2h.elf: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by ./2s2h.elf)
./2s2h.elf: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by ./2s2h.elf)
./2s2h.elf: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by ./2s2h.elf)
./2s2h.elf: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by ./2s2h.elf)
./2s2h.elf: /lib64/libc.so.6: version `GLIBC_2.36' not found (required by /mnt/SDCARD/Data/ports/soh2/libs/libzip.so.5)
./2s2h.elf: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /mnt/SDCARD/Data/ports/soh2/libs/libspdlog.so.1.10)
./2s2h.elf: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /mnt/SDCARD/Data/ports/soh2/libs/libspdlog.so.1.10)
./2s2h.elf: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /mnt/SDCARD/Data/ports/soh2/libs/libspdlog.so.1.10)
./2s2h.elf: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /mnt/SDCARD/Data/ports/soh2/libs/libfmt.so.9)
./2s2h.elf: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /mnt/SDCARD/Data/ports/soh2/libs/libcrypto.so.3)
Running in Fake Keyboard mode
Using ConfigFile soh2.gptk
Joystick 0 has game controller name 'Xbox 360 Controller'

**Comment by cizia64** [comment]
Yes, the problem comes from the old libc that we have with the native firmware. Knulli bypass the native firmware so it has more up to date libc.
In the PortMaster folder of SOH2, you'll find a compatible binary, SOH works on crossmix with the compatibility elf but also has performance issues.

**Comment by sergiogbrox** [comment]
Okay. Sorry for asking for more information, but I'm autistic and sometimes I can't deduce everything clearly hehe. In this case, then it's not possible to update this libc to have the same performance as in Knulli, right? If so, then I can close this issue because there won't be a solution, right? =(

Don't worry, even if there's no way to solve it, I understand that it's a limitation of the native system and I really appreciate your attention and all your effort with this TOP system! Thanks ^^

**Comment by bideg** [comment]
> SOH2

Where exactly can I obtain the compatible SOH2 binary? I searched in the port zip downloaded from the Portmaster website, but I couldn't find it there."
